File: library/src/main/java/com/yzyfdf/library/base/BasePresenter.java
package com.yzyfdf.library.base;

import android.content.Context;

import com.yzyfdf.library.rx.RxManager;

import es.dmoral.toasty.Toasty;

public abstract class BasePresenter<T,E>{
    public Context mContext;
    public E mModel;
    public T mView;
    public RxManager mRxManager = new RxManager();

    public void setVM(T v, E m) {
        this.mView = v;
        this.mModel = m;
        this.onStart();
    }
    public void onStart(){
    }
    public void onDestroy() {
        mRxManager.clear();
    }

    protected void showErrorTip(String msg) {
        Toasty.error(mContext, msg).show();
    }
}

File: library/src/main/java/com/yzyfdf/library/base/BasePresenter.kt
package com.yzyfdf.library.base

import android.content.Context
import com.yzyfdf.library.rx.RxManager

import es.dmoral.toasty.Toasty

abstract class BasePresenter<T, E> {
    var mContext: Context? = null
    var mModel: E? = null
    var mView: T? = null
    var mRxManager = RxManager()

    fun setVM(v: T, m: E) {
        this.mView = v
        this.mModel = m
        this.onStart()
    }

    open fun onStart() {}

    fun onDestroy() {
        mRxManager.clear()
    }

    protected fun showErrorTip(msg: String) {
        Toasty.error(mContext!!, msg).show()
    }
}
